cdap-proto/src/main/java/co/cask/cdap/proto/PluginInstanceDetail.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.proto;
+
+import co.cask.cdap.api.plugin.Plugin;
+
+/**
+ * Represents a plugin instance in an HTTP Response
+ */
+public class PluginInstanceDetail {
+  private final String pluginId;
+  private final Plugin plugin;
```

Comment:
this may be a weird property name, as the entire object represents a plugin. How about 'spec' or something like that?